plugins/package-managers/node/src/main/kotlin/Npm.kt

```diff
@@ -246,8 +240,13 @@ open class Npm(
 
         logger.info { "Searching for 'package.json' files in '$nodeModulesDir'..." }
 
-        val nodeModulesFiles = nodeModulesDir.walk().maxDepth(modulesSearchDepth).filter {
-            it.name == "package.json" && isValidNodeModulesDirectory(nodeModulesDir, nodeModulesDirForPackageJson(it))
+        val visitedDirs = mutableSetOf<File>()
+
+        val nodeModulesFiles = nodeModulesDir.walk().onEnter { dir ->
+            !dir.isSymbolicLink() || visitedDirs.add(dir.realFile())
+        }.filter {
+            it.isFile && it.name == "package.json" &&
+                isValidNodeModulesDirectory(nodeModulesDir, nodeModulesDirForPackageJson(it))
```

Comment:
It could be just me: it seems the area of code is complicated and there I believe it would be worth making a separate preceding commit for the `it.IsFile` check, just minimize the diff of the bug fix.